### PR TITLE
Handle proper exception on missing task

### PR DESCRIPTION
**Pull request by jiivan**
Fix small issue during handshake error
```
2019-07-15 09:09:13 CRITICAL twisted                             Unhandled error in Deferred:
2019-07-15 09:09:13 CRITICAL twisted                             
Traceback (most recent call last):
  File "C:\BuildbotWorker\buildpackage_windows\build\.venv\lib\site-packages\twisted\internet\defer.py", line 653, in _runCallbacks
  File "C:\BuildbotWorker\buildpackage_windows\build\golem\resource\hyperdrive\resourcesmanager.py", line 310, in error_wrapper
  File "C:\BuildbotWorker\buildpackage_windows\build\golem\resource\base\resourceserver.py", line 146, in _download_error
  File "C:\BuildbotWorker\buildpackage_windows\build\golem\client.py", line 656, in resource_failure
  File "C:\BuildbotWorker\buildpackage_windows\build\golem\task\taskcomputer.py", line 135, in resource_failure
  File "C:\BuildbotWorker\buildpackage_windows\build\golem\task\taskcomputer.py", line 415, in __task_finished
  File "C:\BuildbotWorker\buildpackage_windows\build\golem\task\taskkeeper.py", line 679, in task_ended
KeyError: 'e643f64c-a6df-11e9-8020-c555375e7110'
```

### CODE REVIEW

File: golem/task/taskkeeper.py [hunk 1]
```diff
@@ -677,7 +677,7 @@ def task_started(self, task_id):
     def task_ended(self, task_id):
         try:
             self.running_tasks.remove(task_id)
-        except ValueError:
-            logger.warning("Can not remove running task, already removed. "
+        except KeyError:
+            logger.debug("Can not remove running task, already removed. "
```
Comment (maaktweluit) [inline golem/task/taskkeeper.py]: Not sure if i agree this can be a DEBUG log, since it should never happen that this function is called twice.

**Review by maaktweluit — APPROVED** [review]
LGTM!

Small error on the unittests, probably the indent changed from `warning` -> `debug`

File: golem/task/taskkeeper.py (hunk 1 above)
Comment (jiivan) [reply golem/task/taskkeeper.py]: I don't think it's called twice. It's probably called before task is added to `taskkeeper`. Possibly during handshake.

Comment (maaktweluit) [reply golem/task/taskkeeper.py]: ok, in that case its fine to decrease the severity :)

Comment (shadeofblue) [reply golem/task/taskkeeper.py]: @jiivan actually, you could have updated the typo: `Can not` -> `Cannot` if you're changing the line anyway ;)

Comment (jiivan) [reply golem/task/taskkeeper.py]: @shadeofblue okidoki, it should be perfect now :)

**Pull request merged**